src/main/java/net/dries007/tfc/world/classic/fluids/FluidThirstRegistry.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Work under Copyright. Licensed under the EUPL.
+ * See the project README.md and LICENSE.txt for more information.
+ */
+
+package net.dries007.tfc.world.classic.fluids;
+
+import static net.dries007.tfc.Constants.GSON;
+import static net.dries007.tfc.api.util.TFCConstants.MOD_ID;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.apache.commons.io.FileUtils;
+
+import com.google.common.base.Strings;
+import com.google.gson.JsonParseException;
+import com.google.gson.reflect.TypeToken;
+
+import net.dries007.tfc.TerraFirmaCraft;
+import net.dries007.tfc.world.classic.worldgen.vein.VeinRegistry;
+
+public enum FluidThirstRegistry
```

Comment:
Claycorp already said this - this entire config is way way way too complicated and unnecessary. At most, what needs to be configurable is the possible fluids, which can be done via `String[]` in `ConfigTFC`, i.e. "ceramicJugDrinkableFluids". There's no need for a per-fluid config setting.
TL;DR. Remove all of this.